controllers/components/trustyai/trustyai_controller.go

```diff
@@ -55,8 +57,17 @@ func (s *componentHandler) NewComponentReconciler(ctx context.Context, mgr ctrl.
 				handlers.ToNamed(componentApi.TrustyAIInstanceName)),
 			reconciler.WithPredicates(
 				component.ForLabel(labels.ODH.Component(LegacyComponentName), labels.True)),
+			reconciler.WithPredicates(
```

Comment:
I don't know all the details, but it seems that if you don't explicit use an `predicate,Or`,  then by default it is an `And`